lib/move/move.js

```diff
@@ -1,76 +1,59 @@
 'use strict'
 
-const fs = require('graceful-fs')
+const fs = require('../fs')
 const path = require('path')
-const copy = require('../copy').copy
-const remove = require('../remove').remove
-const mkdirp = require('../mkdirs').mkdirp
-const pathExists = require('../path-exists').pathExists
+const { copy } = require('../copy')
+const { remove } = require('../remove')
+const { mkdirp } = require('../mkdirs')
+const { pathExists } = require('../path-exists')
 const stat = require('../util/stat')
 
-function move (src, dest, opts, cb) {
-  if (typeof opts === 'function') {
-    cb = opts
-    opts = {}
-  }
+async function move (src, dest, opts = {}) {
+  const overwrite = opts.overwrite || opts.clobber || false
 
-  opts = opts || {}
+  const { srcStat, isChangingCase = false } = await stat.checkPaths(src, dest, 'move', opts)
 
-  const overwrite = opts.overwrite || opts.clobber || false
+  await stat.checkParentPaths(src, srcStat, dest, 'move')
 
-  stat.checkPaths(src, dest, 'move', opts, (err, stats) => {
-    if (err) return cb(err)
-    const { srcStat, isChangingCase = false } = stats
-    stat.checkParentPaths(src, srcStat, dest, 'move', err => {
-      if (err) return cb(err)
-      if (isParentRoot(dest)) return doRename(src, dest, overwrite, isChangingCase, cb)
-      mkdirp(path.dirname(dest), err => {
-        if (err) return cb(err)
-        return doRename(src, dest, overwrite, isChangingCase, cb)
-      })
-    })
-  })
-}
+  // If the parent of dest is not root, make sure it exists before proceeding
+  const destParent = path.dirname(dest)
+  const parsedParentPath = path.parse(destParent)
+  if (parsedParentPath.root !== destParent) {
+    await mkdirp(path.dirname(dest))
```

Comment:
We already have the dirname, let's use it
```suggestion
    await mkdirp(destParent)
```

Reply:
Fixed in [`7ab00d5` (#1025)](https://github.com/jprichardson/node-fs-extra/pull/1025/commits/7ab00d58bf1f018aaa2f3490e52c5fa401c7c449).